rich/rule.py

```diff
@@ -56,12 +64,18 @@ def __rich_console__(
             title_text.plain = title_text.plain.replace("\n", " ")
             title_text = title_text.tabs_to_spaces()
             rule_text = Text(end=self.end)
-            center = (width - cell_len(title_text.plain)) // 2
-            rule_text.append(character * (center - 1) + " ", self.style)
+            side_width = (width - cell_len(title_text.plain)) // 2
+            if chars_len == 1:
+                side = Text(characters * side_width)
+            else:
+                side = Text(characters * (side_width // (chars_len - 1)))
+            side.truncate(side_width - 1)
+            rule_text.append(str(side) + " ", self.style)
             rule_text.append(title_text)
-            rule_text.append(
-                " " + character * (width - cell_len(rule_text.plain) - 1), self.style
-            )
+            rule_text.append(" " + str(side), self.style)
+            if len(rule_text) < width:
+                rule_text.append(characters[0], self.style)
```

Comment:
This could lead to the last char not being in sequence. i.e. if characters is "abc", the sequence may not always end in an "a".

How about building the left and right text separately. That way you can calculate the length of the right rule like this:

```python
right_length = width - cell_len(left.plain) - cell_len(title.plain)
```

So you wouldn't have to compensate for an odd width.

Reply:
Will do this next